### ISSUE TITLE
Examples don't work

### ISSUE THREAD

**Issue by skx** [riicchhaarrd/cidscropt#2]
There are (currently) three examples:

* hello_world.osc
     * This works fine.

* write_file_stuff.osc
     * This fails because it tries to call the built-in function `fclose` which isn't setup in `src_defaultlib.c`.
     * Commenting out the calls to `fclose()` allows it to run.

* color_circles.osc
     * This tries to call the function `rand()` which isn't defined in `src_defaultlib.c`.
     * Changing the calls to `rand()` to read `randomint()` allos it to get further.
     * But even with that  we see failures:

      deagol ~/cidscropt $ ./bin/cidscropt examples/color_circles.osc
      variable 'vid_width' does not exist!
      BLOCK=1,curpos=832,at=895

Shuffling the variables around so each is defined in the functions eventually allows it to run, but nothing useful happens:

      deagol ~/cidscropt $ ./bin/cidscropt examples/color_circles.osc
      built-in function 'set_pixel' does not exist! (0)
      Segmentation fault

It looks like variables defined inside functions aren't accessible outside that function, which is is what you'd expect from a C-like function.  This is demonstrated by this example:

     function bar()
     {
        global = 3
        print("Global set to " + global + "\n" );
     }

     main() {
        bar()
         //   print("Global is now dead: " + global + "\n")
     }

Uncomment the second print and you see a failure - and this is exactly how the code in the `color_circles` example is setup  - specifically you'll see the function `rand_col()` defined variables, and the `vid_size`, etc, variables in main aren't available in `clear_screen()`.

In short it would be nice if the examples all worked :)

**Comment by riicchhaarrd** [comment]
The hello world I quickly made just to show an up-to-date example, the other examples e.g the color one is from a old interpreter version which is on my github aswell but I still had them, the fclose I changed later to automatically close but I forgot to update the example. I will try to add more examples and fix the old ones that don't apply anymore.

As for globals, there are none except level for now. (In the old interpreter everything was a global)
level is a global object which you can assign fields to.

I also temporarily removed some windows dependent code e.g set_pixel and will readd them later again.

**Comment by skx** [comment]
> I will try to add more examples and fix the old ones that don't apply anymore.

Great, thanks.

Having examples (&documentation!) makes the project use more obvious, and shows what it can do, but even without things being current I found out how it worked well enough with a bit of trial and error.  Good project :)
